### PR TITLE
Add ignore elements flow operator

### PR DESCRIPTION
**Pull request by shariarriday**
Relates https://github.com/actor-framework/actor-framework/issues/1389

Implements ignore elements.

### CODE REVIEW

File: libcaf_core/caf/flow/step/ignore_elements.hpp [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+  explicit ignore_elements() {
+    // nop
+  }
+
+  ignore_elements(ignore_elements&&) = default;
+  ignore_elements(const ignore_elements&) = default;
+  ignore_elements& operator=(ignore_elements&&) = default;
+  ignore_elements& operator=(const ignore_elements&) = default;
+
```
Comment (Neverlord) [inline libcaf_core/caf/flow/step/ignore_elements.hpp:29]: This has no state: rule of 0.

```suggestion
```

File: libcaf_core/caf/flow/step/ignore_elements.hpp [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  template <class Next, class... Steps>
+  bool on_next(const input_type& item, Next& next, Steps&... steps) {
```
Comment (Neverlord) [inline libcaf_core/caf/flow/step/ignore_elements.hpp:31]: Don't you get warnings for unused parameters on MSVC here? If not, we need to figure out the appropriate flags.

```suggestion
  bool on_next(const input_type&, Next&, Steps&...) {
```

File: libcaf_core/caf/flow/step/ignore_elements.test.cpp [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+      .ignore_elements()
+      .take(5)
+      .for_each([&](const int& result_observable) {
+        result.emplace_back(result_observable);
+      });
+  }
+  ctx->run();
+  check_eq(result.size(), 0u);
+}
+
```
Comment (Neverlord) [inline libcaf_core/caf/flow/step/ignore_elements.test.cpp:51]: I think you made these analogous to the `take_last` tests. Good idea in general, but for this operator, there is no edge case to check here. Hence, this is redundant and adds no extra coverage. I'd say we can just drop it.

```suggestion
```

File: libcaf_core/caf/flow/step/ignore_elements.hpp [hunk 4]
```diff
@@ -0,0 +1,46 @@
+// This file is part of CAF, the C++ Actor Framework. See the file LICENSE in
+// the main distribution directory for license terms and copyright or visit
+// https://github.com/actor-framework/actor-framework/blob/master/LICENSE.
+
+#pragma once
+
+#include "caf/detail/ring_buffer.hpp"
+#include "caf/fwd.hpp"
+
+#include <cstddef>
```
Comment (Neverlord) [inline libcaf_core/caf/flow/step/ignore_elements.hpp:10]: Header hygiene: we shouldn't include unnecessary headers.

```suggestion
#include "caf/fwd.hpp"
```

**Review by Neverlord — CHANGES_REQUESTED** [review]

File: libcaf_core/caf/flow/step/ignore_elements.hpp (hunk 2 above)
Comment (shariarriday) [reply libcaf_core/caf/flow/step/ignore_elements.hpp]: I don't see any warning for unused parameters. I will look into appropriate flags for MSVC for this.
